ephios/core/views/workinghour.py

```diff
@@ -42,23 +43,45 @@ def _get_target_user(self):
         return get_object_or_404(UserProfile, pk=self.kwargs["pk"])
 
 
-class DateFilterForm(forms.Form):
+class WorkingHourFilterForm(forms.Form):
     start = forms.DateField(required=False, label=_("From"), widget=CustomDateInput)
     end = forms.DateField(required=False, label=_("To"), widget=CustomDateInput)
+    type = forms.ModelChoiceField(
+        label=_("Event type"),
+        queryset=EventType.objects.all(),
+        required=False,
+        widget=Select2Widget(
+            attrs={
+                "data-placeholder": _("Event type"),
+                "classes": "w-auto",
+            }
+        ),
+    )
 
 
 class WorkingHourOverview(CustomPermissionRequiredMixin, TemplateView):
     template_name = "core/workinghours_list.html"
     permission_required = "core.view_userprofile"
 
-    def _get_working_hours_stats(self, start: Optional[date], end: Optional[date]):
-        participations = (
-            LocalParticipation.objects.filter(
-                state=LocalParticipation.States.CONFIRMED,
-                start_time__date__gte=start,
-                end_time__date__lte=end,
+    def _get_working_hours_stats(
+        self, start: Optional[date], end: Optional[date], type: Optional[EventType]
```

Comment:
could you fix the type annotations here? start and end don't seem optional to me